src/instamatic/calibrate/calibrate_stage_rotation.py

```diff
@@ -0,0 +1,395 @@
+    simulator show that more points offer better accuracy (times in ms):
+
+    -  30 calib. points: pace 1200+/-15, windup -12+/-116, delay 62+/-18
+    -  60 calib. points: pace 1202+/-04, windup -22+/-031, delay 51+/-06
+    - 120 calib. points: pace 1198+/-03, windup   2+/-020, delay 53+/-04
+
+    By default, this function will try testing for rotation speeds of
+    [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12] i.e. JEOL settings first, and
+    [0,1. 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0] i.e. FEI settings after.
+    However, it is worth noting that a microscope might technically support
+    every speed setting above while in practice introduce linear changed
+    only in its limited range, e.g. up to 0.25, as noted on a FEI Tecnai.
+    It is advised to look for irregularities during calibration and limit
+    the calibration only to the range of speeds that will be used in practice.
+
+    ctrl: instance of `TEMController`
+        contains tem + cam interface
+    alpha_spans: `Optional[Sequence[float]]`
+        Alpha rotations whose speed will be measured. Default: range(1, 11, 1).
+    speed_range: `Optional[Sequence[Union[float, int]]]`
+        Spead range to measure. Default: range(1, 13, 1) or range(.1, 1.1, .1).
+    outdir: `str` or None
+        Directory where the final calibration file will be saved.
+
+    return:
+        instance of `CalibStageRotation` class with conversion methods
+    """
+
+    alpha_spans = np.array(alpha_spans or np.array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]))
+    alternating_ones = np.ones(len(alpha_spans)) * (-1) ** np.arange(len(alpha_spans))
+    alpha_targets = np.cumsum(alpha_spans * alternating_ones)
+
+    calib_points: List[SpanSpeedTime] = []
+    starting_stage_alpha = ctrl.stage.a
+    starting_stage_speed = ctrl.stage.get_rotation_speed()
+    ctrl.cam.block()
+    try:
+        try:
+            ctrl.stage.set_rotation_speed(12)
+            assert ctrl.stage.get_rotation_speed() == 12
+        except AssertionError:  # or any other raised if speed can't be set
+            speed_range_default = np.linspace(0.01, 0.2, num=20)
+            speed_options = FEI_ROTATION_SPEED_OPTIONS
+        else:
+            speed_range_default = np.arange(1, 13, step=1)
+            speed_options = JEOL_ROTATION_SPEED_OPTIONS
+        speed_range = speed_range or speed_range_default
+
+        total = len(speed_range) * len(alpha_spans)
+        log(f'Starting rotation speed calibration based on {total} points.')
+
+        with tqdm(total=total) as progress_bar:
+            for speed in speed_range:
+                with ctrl.stage.rotation_speed(speed=float(speed)):
+                    ctrl.stage.a = 0.0
+                    for at, as_ in zip(alpha_targets, alpha_spans):
+                        with timer() as t:
+                            ctrl.stage.a = float(at)
+                        calib_points.append(SpanSpeedTime(as_, speed, t()))
```

Comment:
Although the contextmanager for the timer is clever, I think this is way easier to read:

```suggestion
                        t1 = time.perf_counter()
                        ctrl.stage.a = float(at)
                        t2 = time.perf_counter()
                        calib_points.append(SpanSpeedTime(span, speed, t2-t1))
```